openfl/component/envoy/envoy.py

```diff
@@ -46,20 +54,21 @@ def run(self):
         """Run of the envoy working cycle."""
         while True:
             try:
-                # Workspace import should not be done by gRPC client!
-                experiment_name = self.director_client.get_experiment_data()
+                experiment_name, client_id = self.director_client.wait_experiment()
+                experiment_work_dir = Path(f'{os.getcwd()}/{client_id}/{experiment_name}')
+                data_stream = self.director_client.get_experiment_data(experiment_name)
+                self.create_workspace(experiment_name, experiment_work_dir, data_stream)
```

Comment:
This is a strange signature for the `create_workspace` method. Why it accepts data_stream? Its only responsibility should be to create the workspace, not reading the stream.